crypto/modes/siv128.c

```diff
@@ -0,0 +1,372 @@
+/*
+ * Copyright 2017 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the OpenSSL license (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ *
+ * Copyright 2017, Akamai Technologies. All Rights Reserved.
+ * This code is distributed under the terms of the OpenSSL license
+ * Based on https://github.com/dfoxfranke/libaes_siv
+ */
+
+#include <string.h>
+#include <stdlib.h>
+#include <openssl/crypto.h>
+#include <openssl/cmac.h>
+#include "modes_lcl.h"
+
+#if !defined(OPENSSL_NO_SIV) && !defined(OPENSSL_NO_CMAC)
```

Comment:
you can make no-siv imply no-cmac or otherwise tie them together in Configure; why test for both?

Reply:
It's the opposite; SIV depends on CMAC, so no-cmac implies no-siv. It's already in Configure, so I can remove the CMAC checks.